src/storage/src/sink/kafka.rs

```diff
@@ -421,8 +412,27 @@ impl KafkaSinkState {
 
         let retry_manager = Arc::new(Mutex::new(KafkaSinkSendRetryManager::new()));
 
-        let producer_context =
-            SinkProducerContext::new(Arc::clone(&metrics), Arc::clone(&retry_manager));
+        let (status_tx, mut status_rx) = mpsc::channel(16);
+
+        let producer_context = SinkProducerContext {
+            metrics: Arc::clone(&metrics),
+            healthchecker: status_tx,
+            retry_manager: Arc::clone(&retry_manager),
+        };
+
+        let healthchecker: Arc<Mutex<Option<Healthchecker>>> = Arc::new(Mutex::new(None));
+
+        {
+            let healthchecker = Arc::clone(&healthchecker);
+            task::spawn(|| "producer-error-report", async move {
```

Comment:
maybe prefix this with "kafka-sink-...", but I also think our task names are all over the place.

Reply:
It's a good idea!